src/components/input/input.tsx

```diff
@@ -0,0 +1,33 @@
+import { useFormContext } from "react-hook-form";
+
+interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {
+  Icon: React.FC;
+  formName: string;
+  className?: string;
+  position?: string;
```

Comment:
eu mudaria o tipo dessa propriedade para um enum para ficar mais fácil quando for usar o componente.